evalml/automl/automl_search.py

```diff
@@ -522,9 +527,46 @@ def search(self, data_checks="auto", show_iteration_plot=True):
 
         best_pipeline = self.rankings.iloc[0]
         best_pipeline_name = best_pipeline["pipeline_name"]
+        self._best_pipeline = self.get_pipeline(best_pipeline['id'])
+        if self._train_best_pipeline:
+            X_train = _convert_to_woodwork_structure(X)
+            y_train = _convert_to_woodwork_structure(y)
+            X_threshold_tuning = None
+            y_threshold_tuning = None
+            if self.optimize_thresholds and self.objective.is_defined_for_problem_type(ProblemTypes.BINARY) and self.objective.can_optimize_threshold:
+                X_train, X_threshold_tuning, y_train, y_threshold_tuning = train_test_split(X_train, y_train, test_size=0.2, random_state=self.random_state)
```

Comment:
Hmm... now that #1597 is merged, we should really move the computation of this threshold tuning split out into the `AutoMLSearch` constructor.

This is fine for now. I filed #1621 to track.

Reply:
Thanks for filing!